### ISSUE TITLE
before_title and after_title tag args in base/siteorigin-widget.class.php

### ISSUE THREAD

**Issue by andrejarh** [siteorigin/so-widgets-bundle#876]
Hi guys!
Only after posting the issue on the Panels rep https://github.com/siteorigin/siteorigin-panels/issues/668 and tracing the issue in the code, I noticed that the source of the issue is within the widget bundle.

This line:
https://github.com/siteorigin/so-widgets-bundle/blob/0c973f0d647cf1537a3717e0b1daafc663bae7e7/base/siteorigin-widget.class.php#L173 And the line after are setting the Widget title HTML tag to none. 
This causes the title of the widget to be rendered without any HTML tag (and can't be styled with CSS).

I think there should be the same HTML tag for widget title, as is already used in siteorigin_widget_preview_widget_action():  https://github.com/siteorigin/so-widgets-bundle/blob/develop/base/inc/actions.php#L44 and in the siteorigin_widget_shortcode() here: https://github.com/siteorigin/so-widgets-bundle/blob/develop/base/inc/shortcode.php#L35

I think the solution is to change the before_title and after_title args in base/siteorigin-widget.class.php from 
```
		$args = wp_parse_args( $args, array(
			'before_widget' => '',
			'after_widget' => '',
			'before_title' => '',
			'after_title' => '',
		) );
```

to: 
```

		$args = wp_parse_args( $args, array(
			'before_widget' => '',
			'after_widget' => '',
			'before_title' => '<h3 class="widget-title">',
			'after_title' => '</h3>',
		) );

```

**Comment by braamgenis** [comment]
Hi @andrejarh These are just defaults, not overrides. It's up to the context in which widgets are rendered to decide how widget titles should be displayed. As you pointed out, the shortcode, widget previews and SiteOrigin Page Builder all add `<h3>` tags. A theme's sidebars may output different wrappers for titles. Granted, the empty defaults may not have been the best choice, but given the [length of time](https://github.com/siteorigin/so-widgets-bundle/blame/0c973f0d647cf1537a3717e0b1daafc663bae7e7/base/siteorigin-widget.class.php#L170) it's been this way in released versions I'm opposed to changing it in the base widget class, as the potential benefit doesn't justify the potential cost.

The correct solution in this case is to pass in the desired arguments from the SiteOrigin Widget Block [widget rendering](https://github.com/siteorigin/so-widgets-bundle/blob/1.15.7/compat/block-editor/widget-block.php#L85) code, which I've done and will push shortly. Hopefully, this will resolve any related issues you were having.

**Comment by braamgenis** [comment]
Fixed in fadc8f956ad0a337f91c48b771a5c7d96ad98f75
